doc-content/enterprise-only/springboot/creating-self-contained-image-proc.adoc

```diff
@@ -0,0 +1,98 @@
+[id='creating-self-contained-image-proc_{context}']
+= Creating a self-contained {PRODUCT} Spring Boot JAR file
+
+You can create a self-contained {PRODUCT} Spring Boot JAR file that contains multiple versions of the same KJAR file, including modules. The KJAR files are separated from the `BOOT-INF/lib` JAR files to avoid class loader collisions. Each classpath container KJAR file is isolated for other KJAR files and does not rely on the Spring Boot class loader.
+
+.Prerequisites
+* You have a {PRODUCT} Spring Boot project.
+
+.Procedure
+. Add the following properties to the `service/source/main/resources/application.properties` file of your Spring Boot application:
++
+[source]
+----
+kieserver.classPathContainer=true
+kieserver.autoScanDeployments=true
+----
++
+When both of these properties are set to `true`, Maven automatically scans for container IDs that contain `<artifactId>` and `<version>` at build time.
```

Comment:
not maven, spring boot kie server
and it scans for deployable artifacts that were packaged within spring boot application using
package-dependencies-kjar maven goal of kie-maven-plugin